wasm/lib/src/convert.rs

```diff
@@ -130,10 +132,9 @@ pub fn py_to_js(vm: &VirtualMachine, py_obj: PyObjectRef) -> JsValue {
             view.slice(0, bytes.len() as u32).into()
         }
     } else {
-        match vm.serialize(&py_obj) {
-            Ok(json) => js_sys::JSON::parse(&json).unwrap_or(JsValue::UNDEFINED),
-            Err(_) => JsValue::UNDEFINED,
-        }
+        rustpython_vm::ser_de::PyObjectSerializer::new(vm, &py_obj)
```

Comment:
Instead of creating the new struct `PyObjectSerializer`, consider creating a function `serialize(vm, &py_obj, &serde_wasm_bindgen::Serializer::new())`. It's probably best if `PyObjectSerializer` is a private struct which does not leave the `ser_de` module.

Reply:
I think it's better to do both, because when we use `serde::to_string()` in `json`, it expects a type that's `Deserialize`, and the alternative is to manually pass an `&mut String` to `serde::Serializer::new()`, which seems uglier.